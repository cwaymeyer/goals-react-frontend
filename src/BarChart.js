import {
  Chart as ChartJS,
  CategoryScale,
  LinearScale,
  BarElement,
  Title,
  Tooltip,
  Legend,
} from "chart.js";
import { Bar } from "react-chartjs-2";

function BarChart({ ormPercentage, datePercentage }) {
  ChartJS.register(
    CategoryScale,
    LinearScale,
    BarElement,
    Title,
    Tooltip,
    Legend
  );

  const options = {
    responsive: true,
    plugins: {
      legend: {
        position: "top",
      },
    },
    scales: {
      y: {
        beginAtZero: true,
        max: 100,
      },
    },
  };

  const labels = ["% completed from starting 1RM"];

  const data = {
    labels,
    datasets: [
      {
        label: "Current 1RM",
        data: [ormPercentage],
        backgroundColor: "rgba(255, 99, 132)",
      },
      {
        label: "Time passed",
        data: [datePercentage],
        backgroundColor: "rgba(53, 162, 235)",
      },
    ],
  };

  return <Bar options={options} data={data} />;
}

export default BarChart;
