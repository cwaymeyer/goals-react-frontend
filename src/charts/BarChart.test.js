import { render } from "@testing-library/react";
import BarChart from "./BarChart";

global.ResizeObserver = require("resize-observer-polyfill");

it("renders without crashing", function () {
  render(<BarChart />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<BarChart />);
  expect(asFragment()).toMatchSnapshot();
});
