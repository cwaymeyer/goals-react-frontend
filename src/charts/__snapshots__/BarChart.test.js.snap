// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <canvas
    height="0"
    role="img"
    style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
    width="0"
  />
</DocumentFragment>
`;
